module/input/Camera/src/Camera.cpp

```diff
@@ -327,8 +345,46 @@ namespace input {
                 msg->camera_id = context->camera_id;
                 msg->name      = context->name;
                 msg->timestamp = NUClear::clock::time_point(nanoseconds(ts));
-                // TODO attach Hcw once we have access to sensors
+
+                Eigen::Affine3d Hcw;
+
+                /* Mutex Scope */ {
+                    std::lock_guard<std::mutex> lock(reactor.sensors_mutex);
+
+                    Eigen::Affine3d Hcp = context->Hcp;
+                    Eigen::Affine3d Hpw;
+                    if (reactor.Hpws.empty()) {
+                        Hpw = Eigen::Affine3d::Identity();
+                    }
+                    else {
+                        // Find the first time that is not less than the target time
+                        auto Hpw_it = std::lower_bound(reactor.Hpws.begin(),
+                                                       reactor.Hpws.end(),
+                                                       std::make_pair(msg->timestamp, Eigen::Affine3d::Identity()),
+                                                       [](const auto& a, const auto& b) { return a.first < b.first; });
```

Comment:
Pretty sure we can lose the `make_pair` here
```
auto Hpw_it = std::lower_bound(reactor.Hpws.begin(),
                                                       reactor.Hpws.end(),
                                                       msg->timestamp,
                                                       [](const auto& a, const NUClear::clock::time_point& b) { return a.first < b; });
```

Reply:
I've often had trouble with these signatures... but maybe this time it'll be different!